internal/pkg/service/common/httpserver/middleware/filter.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+)
+
+const requestIgnoredCtxKey = ctxKey("request-ignored")
+
+// FilterFn is a predicate used to determine whether a given http.request should
+// be logged/traced. A Filter must return true if the request should be logged/traced.
+type FilterFn func(*http.Request) bool
+
+func Filter(cfg Config) Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			for _, f := range cfg.filters {
+				if !f(req) {
+					req = req.WithContext(context.WithValue(req.Context(), requestIgnoredCtxKey, true))
+					break
+				}
+			}
+
+			next.ServeHTTP(w, req)
+		})
+	}
+}
+
+// returns true if the request should NOT be logged/traced.
+func isRequestIgnored(req *http.Request) bool {
```

Comment:
Tady sem si akorát všiml, že to začíná malý `i` tady aj jinde používáš velké písmena když tu fci používáš jinde v package.

Reply:
No velke pismeno zanamena ze je to public, tj. da sa pouzit MIMO package, vramci pkg to nehra rolu.
Na `isRequestIgnored` asi nemam zatial use-case, aby sa to pouzivalo mimo `middleware` pkg, ale mohlo by to byt aj public.

Ina situacia je, ze samotny typ je private, tam zvyknem davat metody `Public`, kedze je uz typ private, a ak je private aj jeho konstruktor funkcia `newPrivate() *private`, tak sa neda vytvorit mimo package.
```
type private struct {
}

func (p *private) PublicMethod() {
}
```

Je to ^^^ vyhoda v pripade refactoringu, ked zistim, ze `private` utilitku potrebujem vo viacerych packages, tak zmenim iba `private -> Private` a nemusi prepisovat metody.

^^^^ neviem ci je ten moj koment pochopitelny :) tu je k tomu nieco viac, .... kedy to je a nie je rozdiel: https://stackoverflow.com/questions/37952354/golang-public-method-to-private-struct-is-this-have-any-use-case